augurpy/estimators.py

```diff
@@ -1,25 +1,63 @@
-from typing import Dict, List, Optional, Union
+from dataclasses import dataclass
+from typing import Dict, Literal, Union
 
 from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
 from sklearn.linear_model import LogisticRegression
+from switchlang import switch
+from typing_extensions import TypedDict
+
+
+@dataclass
+class Params(TypedDict, total=False):
+    """Type signature for random forest and logistic regression parameters."""
+
+    n_estimators: int
+    max_depth: int
+    max_features: Union[Literal["auto"], Literal["log2"], Literal["sqrt"], int, float]
+    penalty: Union[Literal["l1"], Literal["l2"], Literal["elasticnet"], Literal["none"]]
+
+
+def _raise_exception():
+    """Raise exception for invalid classifier input."""
+    raise Exception("Missing valid input. Choose rf or lr for categorical labels and rf for continous labels.")
 
 
 def create_estimator(
-    classifier: Union[RandomForestClassifier, RandomForestRegressor, LogisticRegression],
-    rf_params: Optional[List],
-    lr_params: Optional[List],
+    classifier: Union[Literal["rf"], Literal["lr"]],
+    params: Params,
+    classification: bool = True,
 ) -> Union[RandomForestClassifier, RandomForestRegressor, LogisticRegression]:
     """Estimator object.
 
     Args:
         classifier: classifier to use in calculating the area under the curve either random forest or logistic regression
-        rf_params: list of parameters for random forest
-        lr_params: list of parameters for logistic regression
+        params: list of parameters for estimator
 
     Returns:
         Estimator object that classifies.
     """
-    pass
+    definition = classifier + " categorical: " + str(classification)
+    with switch(definition) as c:
+        c.case(
+            "rf categorical: True",
+            lambda: RandomForestClassifier(
+                n_estimators=params.get("n_estimators", 100),
```

Comment:
I'd suggest to document the default parameters maybe in the docstring of this function?

Reply:
Do you mean the docstring of the Params class?

Reply:
No, of the `create_estimator` method I thought.

Reply:
I also noticed that the Params does not do what I had hoped: I would like for it to only let those defined keys be generated and to throw an error or something similar when something undefined is passed.

Reply:
I think the params thing is another topic - ill ask again tomorrow :)

Reply:
Yeah, we can discuss tomorrow. :+1:

Reply:
> No, of the `create_estimator` method I thought.

Is there a specific style I can orient this on?

Reply:
> > No, of the `create_estimator` method I thought.
> 
> Is there a specific style I can orient this on?

Nah. But it's nothing major. Just have an attempt :)